server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java

```diff
@@ -30,6 +33,7 @@
 public class DiscoveryExtensionNode extends DiscoveryNode implements Writeable, ToXContentFragment {
 
     private final PluginInfo pluginInfo;
+    private List<ExtensionDependency> dependencies = Collections.emptyList();
```

Comment:
Why not initialize it with `new ArrayList<>()` and creating immutable list?

Reply:
Aren’t we initializing it back [here](https://github.com/opensearch-project/OpenSearch/pull/5438/files#diff-38a32209c0af2ff74c332ef48b38748b645df72b7f54ea2c8e0b5b6e6f124c20R75) again or am I over thinking?

Reply:
> Aren’t we initializing it back [here](https://github.com/opensearch-project/OpenSearch/pull/5438/files#diff-38a32209c0af2ff74c332ef48b38748b645df72b7f54ea2c8e0b5b6e6f124c20R75) again or am I over thinking?

I mentioned this in another conversation.   We are indeed duplicating things and I expect to refactor eventually to combine the approaches.  Currently the setup is:
 - YAML is read into a list of `ExtensionsSettings.Extension`
 - During initialization the `ExtensionManager` iterates that list and initializes `DiscoveryExtensionNode` for each one copying over the details

The initialization of an empty list is mandatory for the YAML import.  The second one is just copying.

It still goes down to a style question of using an immutable constant vs. instantiating objects and reserving memory for an array backing the list for every extension read in, even if they will never be used.  So for now I prefer this approach. If later we decide to add a feature of on-the-fly adding of dependencies we can modify to make the list mutable.